modules/programs/tmux.nix

```diff
@@ -0,0 +1,123 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.tmux;
+
+  tmuxinatorModule = types.submodule {
+    enable = mkEnableOption "Tmuxinator";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.tmuxinator;
+      defaultText = "pkgs.tmuxinator";
+      description = "Also install Tmuxinator.";
+    };
+
+    workspaces = mkOption {
+      type = types.either types.path types.lines;
+      default = home.file.".tmuxinator/";
+      description = "where tmuxinator templates files are located.";
+    };
+  };
+
+  tmuxpModule = types.submodule {
+    enable = mkEnableOption "Tmuxp";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.tmuxp;
+      defaultText = "pkgs.tmuxp";
+      description = "Also install Tmuxp.";
+    };
+
+    # Need to figure out more about how tmuxp works for this:
+    #
+    # extraConfig = mkOption {
+    #   type = types.lines; # types.either types.path types.lines;
+    #   description = "Additional configuration for the associated plugin.";
+    #   default = "";
+    # };
+  };
+
+  pluginModule = types.submodule {
+    plugin = mkOption {
+      type = types.package;
+      description = "Path of the configuration file to include.";
+    };
+    extraConfig = mkOption {
+      type = types.lines; # types.either types.path types.lines;
+      description = "Additional configuration for the associated plugin.";
+      default = "";
+    };
+  };
+in
+{
+  options = {
+    programs.tmux = {
+      enable = mkEnableOption "Tmux";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.tmux;
+        defaultText = "pkgs.tmux";
+        description = "Tmux package to install.";
+      };
+
+      tmuxp = tmuxpModule;
+      tmuxinator = tmuxinatorModule;
+
+      plugins = mkOption {
+        type = types.listOf pluginModule;
+        description = "List of tmux plugins to include in your .tmux.conf. -- default: sensible plugin.";
+        # default = [ { plugin = pkgs.tmuxPlugins.sensible; } ];
+        default = [ ];
+        example = literalExample ''
+        with pkgs; [
+          tmuxPlugins.sensible
+          tmuxPlugins.resurrect
+          tmuxPlugins.continuum
+        ];
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.either types.path types.lines;
```

Comment:
I'd suggest having simply `types.lines` a user wanting to put the extra configuration in a separate file can just use `builtins.readFile` which doesn't add terribly much noise.